carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -536,9 +536,15 @@ namespace cpp_message
         j2735_msgs::DayOfWeek output;
         
         size_t dow_size= message.size;
-        for (auto i = 0; i < dow_size; i++)
+        uint8_t tmp_binary=0;
+        for (auto i = 0; i < dow_size; i++) // size is 1 as 8 bits are sufficient for bit-wise encoding of 7 days 
         {
-            output.dow[i] = message.buf[i];
+            tmp_binary = message.buf[0] >> 1; // 7 days in a week, while there are 8 entries
```

Comment:
Should this message buffer index be i?

Reply:
Based on discussion, the message buf should only have 1 element representing that day of the week. 
So implementation should be revised to not allow/handle exceptions to this